### PR TITLE
33 extend development docs

### PR DESCRIPTION
**Pull request by marrip**
I will extend the docs in the hope that it will become easier for people to start contributing. Please leave some feedback on what kind of additional information you need in order to make developing easier.

### CODE REVIEW

**Comment by tinavisnovska** [comment]
I like the change and reorganisation. What i think is missing is setting up development environment - i know there are many ways to set it up and anyone seeing at least one approach does not need any help with this, i am concerned about people like me, who need a bit of help getting through the easiest path to get this done. Something like:

Install vs code, add extensions for docker and dev containers, install docker desktop (or whatever is the name of that clickable docker  thingy - in case of a computer which allows this), make vs code and docker talking, clone the repo (unless you have already), checkout the branch you want to work with, open the project in vs code, run the code in dev container, use `typer` command like so... to print tsoppy version. (I've probably forgotten half of the steps, just trying to remember what i did then).

Then one will have a working example of how to run a commmand and most likely will be able to extend that to their own code. And even if not, the experience of following through the process will most likely help to nail down whats the issue.

Not saying that this is a necessity, just that i see this as an existing gap in the documentation for now.

**Review by molversmyr — COMMENTED** [review]
I also really like the modularisation of the README information. Feels much more structured and becomes easier to get an overview of what information exists (with links even🙌🏼). I believe it will facilitate the development process for us, ultimately making it easier for all of us to more actively contribute with development.

I second Tina's comment that a "Getting started" guide could be beneficial. Other sections could be how to run the Tsoppy modules using Typer, or alternatively a link to Typer documentation. Additionally, I think a more detailed guide on how to make good test files for unit testing would be nice.🙂

**Comment by marrip** [comment]
Very valid points from both of you, I have started extending the guide with a preparation section. Let me know if it is helpful and if there is more details I should add. I try to link to all relevant docs rather than writing things out as people use different OS and setups. I hope that is ok ☺️

**Comment by marrip** [comment]
I have added sections about how to use the cli (typer) and some information about unit testing. The link to the typer docs is in the package table. Please read through it and help me fill the gaps and rephrase so that it makes sense. Cheers 🙏

File: docs/guides/development.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+If you are contributing to the codebase of tsoppy, we would like for you to read this section to familiarize yourself with the structure and guidelines of the project.
+
+## Preparation
+
+### Development environment setup
+
+We recommend to install and use [**Visual Studio Code**](https://code.visualstudio.com/) for development but there might be other options more suitable if you are already familiar with them and if those support devcontainers. Download the appropriate version of Visual Studio Code and install it on your machine.
+
+In order for you to use the devcontainer in this repository, you need to have [**Docker**](https://docs.docker.com/get-started/get-docker/) installed. Depending on your OS you should either go with Docker Desktop or install [Docker Engine](https://docs.docker.com/engine/install/) for linux distributions. Also, consider the [VS Code docs for setting up Docker](https://code.visualstudio.com/docs/devcontainers/containers#_installation).
```
Comment (molversmyr) [inline docs/guides/development.md]: Add link to Docker Desktop as well?

```suggestion
In order for you to use the devcontainer in this repository, you need to have [**Docker**](https://docs.docker.com/get-started/get-docker/) installed. Depending on your OS you should either go with [Docker Desktop](https://www.docker.com/products/docker-desktop/) or install [Docker Engine](https://docs.docker.com/engine/install/) for linux distributions. Also, consider the [VS Code docs for setting up Docker](https://code.visualstudio.com/docs/devcontainers/containers#_installation).
```

File: docs/guides/development.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    def <subpackage name>(...):
+        # call your function or create your class instance here
+    ```
+
+1. Remember to add [unit tests](#unit-testing).
+1. In general, make things configurable and avoid hard-coding paths and variables that might be subjected to changes.
+
+## Unit testing
+
+We use pytest for unit testing and some kind of table-driven testing in order to reduce biolerplate code. Unit tests should be placed in `tests/<subpackage>_<module>_test.py`. If test data is necessary add it under `tests/test_data/<subpackage>_<module>/<function>/<test case name>.py`. Cover all edge cases as well as use cases from the different nodes.
```
Comment (molversmyr) [inline docs/guides/development.md]: Typo

```suggestion
We use pytest for unit testing and some kind of table-driven testing in order to reduce boilerplate code. Unit tests should be placed in `tests/<subpackage>_<module>_test.py`. If test data is necessary add it under `tests/test_data/<subpackage>_<module>/<function>/<test case name>.py`. Cover all edge cases as well as use cases from the different nodes.
```

**Review by molversmyr — COMMENTED** [review]
Excellent work!👏🏼
We can probably make it more comprehensive if we want to, but I think this is a great starting point to build on as our development progresses.

Two (nitpicky) small remarks below:

**Comment by marrip** [comment]
> Excellent work!👏🏼 We can probably make it more comprehensive if we want to, but I think this is a great starting point to build on as our development progresses.
> 
> Two (nitpicky) small remarks below:

Thanks for the fast review. I agree, we probably have to add more information on the way forward but these gaps will naturally emerge when people start using the docs. ☺️ 

Applied your suggestions 🙏

**Review by molversmyr — APPROVED** [review]

**Comment by danielvo** [comment]
Very nice work! This should make it much easier for a new contributor to start with their first feature. :-D

(The guide should also help the reviewers, as explicit expectations are easier to meet.)

If anyone runs into unforeseen issues when setting up their environment and starting developing, we can update the guide accordingly. :-)

**Comment by marrip** [comment]
> Very nice work! This should make it much easier for a new contributor to start with their first feature. :-D
> 
> (The guide should also help the reviewers, as explicit expectations are easier to meet.)
> 
> If anyone runs into unforeseen issues when setting up their environment and starting developing, we can update the guide accordingly. :-)

could you approve if you approve 😜

**Review by danielvo — APPROVED** [review]

**Comment by marrip** [comment]
thanks @danielvo 🙏
